build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts

```diff
@@ -79,6 +75,52 @@ export abstract class LeafTask extends Task {
 		leafTasks.add(this);
 	}
 
+	public initializeWeight() {
+		if (this.parentWeight === -1) {
+			this.parentWeight = this.computeParentWeight() + this.taskWeight;
+			traceTaskInitWeight(`${this.nameColored}: ${this.parentWeight}`);
+		}
+		return this.parentWeight;
+	}
+
+	private computeParentWeight() {
+		let sum = 0;
+		for (const t of this.parentLeafTasks.values()) {
+			sum += t.taskWeight;
+		}
+		return sum;
+	}
+
+	private get parentLeafTasks() {
+		if (this._parentLeafTasks === undefined) {
+			const parentLeafTasks = new Set<LeafTask>(this.directParentLeafTasks);
+			this._parentLeafTasks = parentLeafTasks;
+			this.directParentLeafTasks
+				.map((task) => task.parentLeafTasks)
+				.forEach((p) => p.forEach((t) => parentLeafTasks.add(t)));
+		}
+		return this._parentLeafTasks;
+	}
+	// End function used by initialWeight phase
```

Comment:
Nit: Does this comment maybe indicate things could be better factored?

Reply:
No.  It's a left over comment.  I will remove.